src/core/ir/eval.rs

```diff
@@ -95,6 +99,97 @@ impl IR {
 
                     Ok(value)
                 }),
+                IR::ModifyInput(input_transforms) => {
+                    if let Some(args) = ctx.path_arg::<&str>(&[]) {
+                        let args: IndexMap<Name, ConstValue> = args
+                            .as_object()
+                            .unwrap()
+                            .iter()
+                            .map(|(name, value)| {
+                                if name.to_string().eq(&input_transforms.arg_name) {
+                                    (
+                                        name.clone(),
+                                        handle_args(
+                                            value,
+                                            input_transforms,
+                                            &input_transforms.arg_type,
+                                        )
+                                        .into_owned(),
+                                    )
+                                } else {
+                                    (name.clone(), value.clone())
+                                }
+                            })
+                            .collect();
+
+                        fn handle_args<'a>(
+                            args: &'a ConstValue,
+                            input_transforms: &'a InputTransforms,
+                            type_of: &'a str,
+                        ) -> Cow<'a, ConstValue> {
```

Comment:
This function should be implemented inside of `JsonLike`. It should be a default implementation.